backend/test/dataset/test_create_dataset.py

```diff
@@ -114,37 +107,9 @@ def test_create_dataset_already_exists(mock_s3, mock_dataset_dao):
     expected_response = generate_html_response(400, error_message)
 
     assert lambda_handler(mock_event, mock_context) == expected_response
-    mock_s3.put_object.assert_not_called()
 
 
-@mock.patch("ml_space_lambda.dataset.lambda_functions.dataset_dao")
-@mock.patch("ml_space_lambda.dataset.lambda_functions.s3")
-def test_create_dataset_client_error(mock_s3, mock_dataset_dao):
```

Comment:
We _could_ still get throttle from our dynamo during our DAO call but in this particular case I think we have sufficient coverage.